src/transaction/transaction.rs

```diff
@@ -232,6 +234,69 @@ impl Transaction {
         .commit()
         .await
     }
+
+    /// Pessimisticly lock the keys
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Config, transaction::Client};
+    /// # use futures::prelude::*;
+    /// # use tikv_client_common::Key;
+    /// # futures::executor::block_on(async {
+    /// # let client = Client::new(Config::default()).await.unwrap();
+    /// let mut txn = client.begin().await.unwrap();
+    /// // ... Do some actions.
+    /// let key1: Key = b"key1".to_vec().into();
+    /// let result: () = txn.pessimistic_lock(vec![key1.clone()]).await.unwrap();
+    /// # });
+    /// ```
+    pub async fn pessimistic_lock(
```

Comment:
I think this should not be part of the public API, instead we should set the transaction to be optimistic or pessimistic, and if pessimistic then we take the lock on each update/insert/delete, and we can add an option for reads to make locking reads.